code/WorkInProgress/AzrunStuff.dm

```diff
@@ -57,5 +59,165 @@
 		proc/init_exponential_smoothing(sample_interval, time_const)
 			init_basic(1.0 - ( eulers ** ( -sample_interval / time_const )))
 
+/// Transformation Manager for Thermo-Electric Generator
+/datum/teg_transformation_mngr
+	var/obj/machinery/power/generatorTemp/generator
+	var/static/list/datum/teg_transformation/possible_transformations
 
+	New(teg)
+		. = ..()
+		generator = teg
+		if(!possible_transformations)
+			possible_transformations = list()
+			for(var/T in childrentypesof(/datum/teg_transformation))
+				var/datum/teg_transformation/TT = new T
+				possible_transformations += TT
 
+	disposing()
+		. = ..()
+		generator = null
+
+	/// Periodic function to check if transformation by reagent is possible
+	proc/check_reagent_transformation()
+		if(generator?.active_form?.skip_transformation_checks) return
+		for(var/datum/teg_transformation/T as() in possible_transformations)
+			if(generator.active_form?.type == T.type) continue // Skip current form
+
+			var/reagents_present = length(T.required_reagents)
+			for(var/R as() in T.required_reagents)
+				if(generator.circ1.reagents.get_reagent_amount(R) + generator.circ2.reagents.get_reagent_amount(R) >= T.required_reagents[R])
+				else
+					reagents_present = FALSE
+					break
+
+			if(reagents_present)
+				SPAWN_DBG(0)
```

Comment:
Why is this spawn necessary?

Reply:
Spawn to offload any animations and sleeps to ensure no interference with machines thread, vs moving spawn.  As there are no current reagent transformations as part of the PR this is not technically necessary at this time.  I have two back-burnered transformations one of which has been soft-reviewed by moonlol for flock.